### DISCUSSION TITLE
Multiple base stations coverage map issue

### DISCUSSION THREAD

**Discussion by pablosreyero** [NVlabs/sionna#295] [Q&A] [answered]
Hi,

I have 2 questions, and both are related to the coverage map.

By studying the source code, and the API documentation, I just realised the fact that we can get the contribution of a certain transmitter, thanks to the `scene.preview(·)` function, by providing the transmitter's name, but, Is there a way we can visualize all contributions (every coverage map associated to every transmitter) at once? I'm afraid that this won't be possible right? I say this, since I just tried to assign a list to `cm_tx` with all transmitter's names on `scene.preview(·)` and I got: `ValueError: Invalid type for "tx": Must be a string or int`. I also checked  the source code to verify it. Nevertheless, any reponse reagarding this matter will be great.

My second question, has to do with a problem regarding the coverage map of all transmitters except the one that was firstly placed. When I take a look at the coverage map with `cm.show(·)` for all transmitters, I see how the first positioned transmitter, covers a greater geographical area, and reaches higher path gains, while for the rest of transmitters, I clearly see how the coverage notably decreases ending in a very poor coverage area and low path gains, in comparison with the first Tx, even though all TXs are oriented towards the receiver. I'm also aware that the `orientation` variable in `Transmitter(·)` is ignored, if `look_at` $\neq 0$, which is exactly my case, since `look_at = ue_position` (UE=Rx). Is there any reason that explains this incredible decrease in gain? Is it related to the scene? or perhaps any property I'm missing in the source code?

Here it is my unfinished and humble piece of code:
[TestCode.zip](https://github.com/NVlabs/sionna/files/13729136/TestCode.zip)

Here are the figures regarding my second question:

Coverage Map of the first Tx:
![image](https://github.com/NVlabs/sionna/assets/104429807/a39ce423-e326-4c05-b50e-5210d3cc29c7)

Coverage Map of the second Tx:
![image](https://github.com/NVlabs/sionna/assets/104429807/0e9ecc5d-2f73-41e3-ad4a-347a47765226)

Coverage Map of the third Tx:
![image](https://github.com/NVlabs/sionna/assets/104429807/a66d79c3-d581-405b-8cec-0ab4e61820e0)

Coverage Map of the fourth Tx:
![image](https://github.com/NVlabs/sionna/assets/104429807/62c4bd08-86a6-415e-99a3-3507a6e25b20)

Coverage Map of the fifth Tx:
![image](https://github.com/NVlabs/sionna/assets/104429807/22603c1f-251f-46bb-a0fc-74308b4e8118)

Coverage Map of the sixth Tx:
![image](https://github.com/NVlabs/sionna/assets/104429807/20da08d7-fd26-4569-b47d-ea7bd69e9ea0)

Coverage Map of the seventh Tx:
![image](https://github.com/NVlabs/sionna/assets/104429807/b968a892-8e4f-48a6-8de5-8b852c84dce4)

Coverage Map of the eighth Tx:
![image](https://github.com/NVlabs/sionna/assets/104429807/e10d7b71-f48a-40e3-8579-e22139849c3b)

Many thanks in advance for your help, and kind regards,

Pablo.-

**Comment by faycalaa** [comment, marked as answer]
Hi,

1. The Sionna API is designed to display the coverage map for a selected transmitter. However, if you need the coverage maps for all transmitters, you can obtain them as a tensor using [CoverageMap.as_tensor()](https://nvlabs.github.io/sionna/api/rt.html#sionna.rt.CoverageMap.as_tensor).

2. In [Scene.coverage_map()](https://nvlabs.github.io/sionna/api/rt.html#sionna.rt.Scene.coverage_map), the `precoding_vec` and `combining_vec` parameters require vectors, not boolean values. To address your issue, try using the default values for these arguments.

**Reply by pablosreyero** [reply]
Ok, got it! 

Thanks.
